pwncat/platform/linux.py

```diff
@@ -494,36 +494,65 @@ class LinuxPath(pathlib.PurePosixPath):
     def readable(self):
         """Test if a file is readable"""
 
-        uid = self._target._id["euid"]
-        gid = self._target._id["egid"]
+        # refresh the uid, to pick up any changes
+        self._target.refresh_uid()
```

Comment:
I think this should actually happen when exiting interactive/remote mode. This would be best called in `def interactive(self, value)` just after updating `self.shell`.

The reason is that during automated enumeration (or some other possible actions like escalation), these calls can happen a lot, and `refresh_uid()` can cause significant performance problems at scale. Also, if automated escalation is used, `refresh_uid()` is automatically called. The only other time the UID would change is if the user does it in remote/interactive mode, so just double checking it during the switch probably makes me most sense.

Reply:
I was actually testing `upload` and `download` and other similar features to first make sure there are no more issues with __file read/write stuff__.

One possible way would be to compare the `pid`, if it changes we make use of `refresh_uid()`
Since it is what we are currently fetching every time we switch between the prompts

Reply:
That's a fair point. We're unlikely to change users without changing processes; at least I can't think of any way that would happen.

Reply:
Even that check should be consolidated to only the escalation framework/command and the interactivity switch logic.

The reason I have that opinion is that we need a standard of when that information goes stale, otherwise we end up constantly repeating that check all over the place, and the cached information becomes almost useless. Not to mention messier code.